### PR TITLE
[LL] - [LIVE-9115] - Add typings to useFeature and improve its overall dev experience

### PR DESCRIPTION
**Pull request by sshmaxime**
### 📝 Description

This PR improves the developer experience of the `useFeature` hook by adding typings to it, it also slightly change few things here and there, all related to the `Feature` concept in our apps.

**Done:**

- Typed `useFeature`
- Renamed `feature` props to `featureId` in FeatureToggle component.
- Renamed `defaultFeatures` to `DEFAULT_FEATURES`.
- Created a norm naming for features: `Feature_X` (Open to changes, I didn't know what else to name it)
- Renamed `recoverFeatueFlag.ts` to `recoverFeatureFlag.ts`

**Notes:**

- I added few `@ts-expect-error` because some functions doesn't accept `undefined` as parameter and there is a need of thinking on what would be the correct to handle that. For example, what do we do if the link passed to `openLink` is undefined ? Because technically as of today it could be `undefined`. That being said I think this is to be done in another PR, this one is already pretty big.
- There is a lot of changed files but it mainly comes for the fact that `useFeature` is used everywhere.

**How to properly add a new Feature Flag:**

- Add its type in `libs/ledgerjs/packages/types-live/src/feature.ts`.
- Add its default value in `libs/ledger-live-common/src/featureFlags/defaultFeatures.ts`.
- Rebuild the lib if necessary: `pnpm build:libs`

### ❓ Context

- **Impacted projects**: `LL` 
- **Linked resource(s)**: [LIVE-9115] 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-9115]: https://ledgerhq.atlassian.net/browse/LIVE-9115?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/screens/Onboarding/steps/welcome.tsx [hunk 1]
```diff
@@ -207,7 +207,9 @@ function OnboardingStepWelcome({ navigation }: NavigationProps) {
           >
             {t("onboarding.stepWelcome.start")}
           </Button>
-          {recoverFeature?.enabled && recoverFeature?.params?.onboardingLogin ? (
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/screens/Onboarding/steps/welcome.tsx:210]: `onboardingLogin` doesn't exist in the current params. I don't know what it was referring to, so I took a guess, correct me if I'm wrong.

File: apps/ledger-live-desktop/src/renderer/modals/Receive/steps/StepReceiveStakingFlow.tsx [hunk 2]
```diff
@@ -88,6 +88,7 @@ const StepReceiveStakingFlow = (props: StepProps) => {
       ...getTrackProperties(),
       link: supportLink,
     });
+    // @ts-expect-error TYPINGS
     openURL(supportLink, "OpenURL", getTrackProperties());
```
Comment (sshmaxime) [inline apps/ledger-live-desktop/src/renderer/modals/Receive/steps/StepReceiveStakingFlow.tsx:92]: `openURL` doesn't accept `undefined`.

File: apps/ledger-live-desktop/src/storyly/useStoryly.tsx [hunk 3]
```diff
@@ -45,14 +45,13 @@ export const useStoryly = (instanceId: StorylyInstanceID) => {
...
-    params: { stories },
-  } = useFeature("storyly") || {};
+  const storyly = useFeature("storyly");
 
   useLayoutEffect(() => {
     ref.current?.init({
       layout: "classic",
-      token: stories[instanceId].token,
+      // @ts-expect-error TYPINGS
+      token: storyly.params?.stories[instanceId].token,
```
Comment (sshmaxime) [inline apps/ledger-live-desktop/src/storyly/useStoryly.tsx:54]: `token` could be undefined.

File: apps/ledger-live-mobile/src/components/RootNavigator/ReceiveFundsNavigator.tsx [hunk 4]
```diff
@@ -108,7 +108,8 @@ export default function ReceiveFundsNavigator() {
               {hasClosedNetworkBanner && (
                 <HelpButton
                   eventButton="Choose a network article"
-                  url={depositNetworkBannerMobile?.params.url}
+                  // @ts-expect-error TYPINGS
+                  url={depositNetworkBannerMobile?.params?.url}
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/components/RootNavigator/ReceiveFundsNavigator.tsx:112]: `HelpButton` doesn't accept `undefined` as `url`.

File: apps/ledger-live-mobile/src/components/RootNavigator/ReceiveFundsNavigator.tsx [hunk 5]
```diff
@@ -193,7 +194,8 @@ export default function ReceiveFundsNavigator() {
             <Flex alignItems="center" justifyContent="center" flexDirection="row">
               {hasClosedWithdrawBanner && (
                 <HelpButton
-                  url={depositWithdrawBannerMobile?.params.url}
+                  // @ts-expect-error TYPINGS
+                  url={depositWithdrawBannerMobile?.params?.url}
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/components/RootNavigator/ReceiveFundsNavigator.tsx:198]: `HelpButton` doesn't accept `undefined` as `url`.

File: apps/ledger-live-mobile/src/components/StorylyStories/StoriesConfig.tsx [hunk 6]
```diff
@@ -24,6 +24,7 @@ const StoriesConfig: React.FC<Props> = ({ instanceID }) => {
   const { overrideFeature } = useFeatureFlags();
   const featureValue = useFeature("storyly");
   const stories = featureValue?.params?.stories;
+  // @ts-expect-error TYPINGS
   const storyConfig = stories[instanceID];
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/components/StorylyStories/StoriesConfig.tsx:28]: `stories` could be undefined.

File: apps/ledger-live-mobile/src/components/StorylyStories/StorylyWrapper.tsx [hunk 7]
```diff
@@ -27,6 +27,7 @@ const StorylyLocalizedWrapper = forwardRef((props: Props, ref: ForwardedRef<Stor
   const [fallbackToEnglish, setFallbackToEnglish] = useState(false);
 
   const {
+    // @ts-expect-error TYPINGS
     params: { stories },
   } = useFeature("storyly") || {};
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/components/StorylyStories/StorylyWrapper.tsx:32]: Trying to destruct `params` which could be undefined.

File: apps/ledger-live-mobile/src/hooks/newsfeed/useCryptopanicPosts.ts [hunk 8]
```diff
@@ -39,7 +39,8 @@ export function useCryptopanicPosts(
           ...params,
           page,
           regions: cryptopanicLocale && [cryptopanicLocale as CryptopanicAvailableRegionsType],
-          auth_token: newsfeedPageFeature?.params.cryptopanicApiKey,
+          // @ts-expect-error TYPINGS
+          auth_token: newsfeedPageFeature?.params?.cryptopanicApiKey,
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/hooks/newsfeed/useCryptopanicPosts.ts:43]: Doesn't like it to be possibly `undefined`.

File: apps/ledger-live-mobile/src/logic/notifications.tsx [hunk 9]
```diff
@@ -153,13 +153,15 @@ const useNotifications = () => {
     }
 
     // minimum accounts number criteria
+    // @ts-expect-error TYPINGS
     const minimumAccountsNumber: number =
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/logic/notifications.tsx:157]: Enforcing type `number` on `number | undefined`

File: apps/ledger-live-mobile/src/screens/Discover/index.tsx [hunk 10]
```diff
@@ -197,17 +197,18 @@ function Discover() {
...
                 subTitle: t("discover.sections.referralProgram.desc"),
                 onPress: () => {
                   readOnlyTrack("referralProgram");
                   track("Discover - Refer Program - OpenUrl", {
-                    url: referralProgramConfig?.params.url,
+                    url: referralProgramConfig?.params?.url,
                   });
-                  Linking.openURL(referralProgramConfig?.params.url);
+                  // @ts-expect-error TYPINGS
+                  Linking.openURL(referralProgramConfig?.params?.url);
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/screens/Discover/index.tsx:211]: `openURL` doesn't like `undefined`.

File: apps/ledger-live-mobile/src/screens/ReceiveFunds/02-SelectNetwork.tsx [hunk 11]
```diff
@@ -156,8 +156,9 @@ export default function SelectNetwork({ navigation, route }: Props) {
       type: "card",
       page: "Choose a network",
     });
-    Linking.openURL(depositNetworkBannerMobile?.params.url);
-  }, [depositNetworkBannerMobile?.params.url]);
+    // @ts-expect-error TYPINGS
+    Linking.openURL(depositNetworkBannerMobile?.params?.url);
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/screens/ReceiveFunds/02-SelectNetwork.tsx:160]: `openURL` doesn't like `undefined`.

File: apps/ledger-live-mobile/src/screens/ReceiveFunds/03-Confirmation.tsx [hunk 12]
```diff
@@ -121,8 +121,9 @@ function ReceiveConfirmationInner({ navigation, route, account, parentAccount }:
       type: "card",
       page: "Receive Account Qr Code",
     });
-    Linking.openURL(depositWithdrawBannerMobile?.params.url);
-  }, [depositWithdrawBannerMobile?.params.url]);
+    // @ts-expect-error TYPINGS
+    Linking.openURL(depositWithdrawBannerMobile?.params?.url);
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/screens/ReceiveFunds/03-Confirmation.tsx:125]: `openURL` doesn't like `undefined`.

File: libs/ledger-live-common/src/hooks/recoverFeatueFlag.ts [hunk 13]
```diff
@@ -63,9 +63,7 @@ export function useUpsellPath(servicesConfig: Feature<DesktopFeature> | null): s
   return usePath(servicesConfig, uri);
 }
 
-export function useLoginURI(
-  servicesConfig: Feature<MobileFeature | DesktopFeature> | null,
-): string | undefined {
+export function useLoginURI(servicesConfig: Feature<DesktopFeature> | null): string | undefined {
```
Comment (sshmaxime) [inline libs/ledger-live-common/src/hooks/recoverFeatueFlag.ts:66]: `account.loginURI` doesn't exist on the feature `protectServicesMobile`. Additionally, this is only use in `Desktop`, so I took the liberty to just keep it for desktop only.

File: apps/ledger-live-mobile/src/screens/Onboarding/steps/welcome.tsx [hunk 14]
```diff
@@ -207,7 +207,9 @@ function OnboardingStepWelcome({ navigation }: NavigationProps) {
...
+          {/* @TODO check if proper URL */}
+          {recoverFeature?.enabled &&
+          recoverFeature?.params?.onboardingRestore.postOnboardingURI ? (
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/screens/Onboarding/steps/welcome.tsx:212]: `onboardingLogin` doesn't exist in the current params. I don't know what it was referring to, so I took a guess, correct me if I'm wrong.

File: libs/ledger-live-common/src/hooks/recoverFeatureFlag.ts [hunk 15]
```diff
@@ -27,52 +27,60 @@ export function usePostOnboardingURI(
...
+  servicesConfig: Feature_ProtectServicesDesktop | null,
+): string | undefined {
   const uri = useUpsellURI(servicesConfig);
 
   return usePath(servicesConfig, uri);
 }
 
 export function useLoginURI(
-  servicesConfig: Feature<MobileFeature | DesktopFeature> | null,
+  servicesConfig: Feature_ProtectServicesDesktop | null,
```
Comment (sshmaxime) [inline libs/ledger-live-common/src/hooks/recoverFeatureFlag.ts:73]: Never used for `MobileFeature` so I removed the type.

**Review by gre — APPROVED** [review]
great work!

**Review by KVNLS — APPROVED** [review]
Great job, this is a great improvement!
I think that we need to fix  `@ts-expect-error` you inserted, could you quickly fix the easy ones and fix the hard ones on a dedicated PR later, please?

**Pull request merged**